modules/component/spec/core/operators/processCdAwareObservables.spec.ts

```diff
@@ -0,0 +1,212 @@
+import { processCdAwareObservables } from '../../../src/core/operators';
+import { EMPTY, Observable, of, pipe, throwError } from 'rxjs';
+import { map, tap } from 'rxjs/operators';
+import { ArgumentNotObservableError } from '../../../src/core/utils';
+
+describe('processCdAwareObservables', () => {
+  it('should work as RxJS operator', () => {
+    const empty: Observable<Observable<any>> = EMPTY;
+    const source: Observable<Observable<any>> = empty.pipe(
+      processCdAwareObservables(o$ => o$, o$ => o$, o$ => o$)
+    );
+    const complete: jasmine.Spy = jasmine.createSpy('complete');
+    source.subscribe({
+      complete: complete,
+    });
+    expect(complete).toHaveBeenCalled();
+  });
+
+  describe('that mirror type safety from toObservableValue', () => {
+    it('should take observables', () => {
+      const observable: Observable<Observable<never>> = of(EMPTY);
+      const complete: jasmine.Spy = jasmine.createSpy('complete');
+      observable
+        .pipe(processCdAwareObservables(o$ => o$, o$ => o$, o$ => o$))
+        .subscribe({
+          complete: complete,
+        });
+      expect(complete).toHaveBeenCalled();
+    });
+
+    it('should take a promise', () => {
+      const observable: Observable<Promise<any>> = of(Promise.resolve(42));
+      observable
+        .pipe(processCdAwareObservables(o$ => o$, o$ => o$, o$ => o$))
+        .subscribe(v => expect(v).toBe(42));
+    });
+
+    it('should take a undefined', () => {
```

Comment:
I think it should be "should take an undefined". It appears in some other places as well.

Reply:
done